PsfLauncher/PsfPowershellScriptRunner.h

```diff
@@ -0,0 +1,305 @@
+
+		return commandString;
+	}
+
+	const std::wstring GetScriptPath(const psf::json_object& scriptInformation) const
+	{
+		//Get throws if the key does not exist.
+		return scriptInformation.get("scriptPath").as_string().wide();
+	}
+
+	DWORD GetTimeout(const psf::json_object& scriptInformation)
+	{
+		DWORD timeout = INFINITE;
+		auto timeoutObject = scriptInformation.try_get("timeout");
+		if (timeoutObject)
+		{
+			//Timout needs to be in milliseconds.
+			//Multiple seconds from config by milliseconds.
+			timeout = (DWORD)(1000 * timeoutObject->as_number().get_unsigned());
+		}
+
+		return timeout;
+	}
+
+	void ThrowIfScriptDoesNotExists(const std::wstring& scriptPath)
+	{
+		std::filesystem::path powershellScriptPath(scriptPath);
+		bool doesScriptExist = false;
+		//The file might be on a network drive.
+		doesScriptExist = std::filesystem::exists(powershellScriptPath);
+
+		//Check on local computer.
+		if (!doesScriptExist)
+		{
+			doesScriptExist = std::filesystem::exists(currentDirectory / powershellScriptPath);
+		}
+
+		THROW_HR_IF_MSG(HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND), !doesScriptExist, "The starting script does not exist");
+	}
+
+	bool GetRunOnce(const psf::json_object& scriptInformation)
+	{
+		bool shouldRunOnce = false;
+		auto runOnceObject = scriptInformation.try_get("runOnce");
+		if (runOnceObject)
+		{
+			shouldRunOnce = runOnceObject->as_boolean().get();
+		}
+
+		return shouldRunOnce;
+	}
+
+	int GetShowWindowAction(const psf::json_object& scriptInformation)
+	{
+		//0 is to minimize
+		int showWindowAction = 0;
+		auto showWindowObject = scriptInformation.try_get("showWindow");
+		if (showWindowObject)
+		{
+			bool showWindow = showWindowObject->as_boolean().get();
+			if (showWindow)
+			{
+				//5 in SW_SHOW.
+				showWindowAction = 5;
+			}
+		}
+
+		return showWindowAction;
+	}
+
+	bool GetRunInVirtualEnvironment(const psf::json_object& scriptInformation)
+	{
+		bool runInVirtualEnvironment = true;
+		auto runInVirtualEnvironmentJObject = scriptInformation.try_get("runInVirtualEnvironment");
+
+		if (runInVirtualEnvironmentJObject)
+		{
+			runInVirtualEnvironment = runInVirtualEnvironmentJObject->as_boolean().get();
+		}
+
+		return runInVirtualEnvironment;
+	}
+
+	bool GetWaitForScriptToFinish(const psf::json_object& scriptInformation)
+	{
+		bool waitForScriptToFinish = false;
+		auto waitForStartingScriptToFinishObject = scriptInformation.try_get("wait");
+		if (waitForStartingScriptToFinishObject)
+		{
+			waitForScriptToFinish = waitForStartingScriptToFinishObject->as_boolean().get();
+		}
+
+		return waitForScriptToFinish;
+	}
+
+	HRESULT CheckIfShouldRun(bool& canScriptRun, const psf::json_object& scriptInformation)
+	{
+		bool shouldRunOnce = GetRunOnce(scriptInformation);
+		canScriptRun = true;
+		if (shouldRunOnce)
+		{
+			std::wstring runOnceSubKey = L"SOFTWARE\\";
+			runOnceSubKey.append(psf::current_package_full_name());
+			runOnceSubKey.append(L"\\runOnce");

```

Comment:
>runOnce [](start = 28, length = 7)

This key is pretty ambiguous, suggest making it something like PSFScriptHasRun or something specific to not clash with potential packaged app keys.